flexibleengine/resource_flexibleengine_obs_bucket_notifications.go

```diff
@@ -0,0 +1,188 @@
+package flexibleengine
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk/openstack/obs"
+)
+
+func resourceObsBucketNotifications() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceObsBucketNotificationCreate,
+		UpdateContext: resourceObsBucketNotificationCreate,
+		ReadContext:   resourceObsBucketNotificationRead,
+		DeleteContext: resourceObsBucketNotificationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"bucket": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"notifications": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"topic_urn": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"events": {
+							Type:     schema.TypeList,
+							Required: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+								ValidateFunc: validation.StringInSlice([]string{
+									"ObjectCreated:*", "ObjectCreated:Put", "ObjectCreated:Post", "ObjectCreated:Copy",
+									"ObjectCreated:CompleteMultipartUpload", "ObjectRemoved:*", "ObjectRemoved:Delete",
+									"ObjectRemoved:DeleteMarkerCreated",
+								}, false),
+							},
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"prefix": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"suffix": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceObsBucketNotificationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*Config)
+	obsClient, err := config.ObjectStorageClient(GetRegion(d, config))
+	if err != nil {
+		return diag.Errorf("Error creating OBS client: %s", err)
+	}
+
+	bucket := d.Get("bucket").(string)
+	notificationOpt := obs.SetBucketNotificationInput{}
+	notificationOpt.Bucket = bucket
+
+	// set notification
+	configurations := buildTopicConfiguration(d)
+	notificationOpt.TopicConfigurations = configurations
+	_, err = obsClient.SetBucketNotification(&notificationOpt)
+	if err != nil {
+		return diag.Errorf("Error setting Notification Configuration of OBS bucket: %s, err: %s", bucket, err)
```

Comment:
```suggestion
		return diag.Errorf("Error setting Notification Configuration of OBS bucket %s, err: %s", bucket, err)
```
the first `:` should be removed

Reply:
OK